.github/workflows/basic.yml

```diff
@@ -0,0 +1,101 @@
+name: Basic build and test workflow
+
+on:
+  push:
+    branches: [ "main", "dev" ]
+  pull_request:
+    branches: [ "main", "dev" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_and_test:
+    name: Compile and test
+
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_DB: clementine
+          POSTGRES_USER: clementine
+          POSTGRES_PASSWORD: clementine
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 50
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - uses: actions/cache@v4
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+    - name: Download Bitcoin
+      run: wget https://bitcoin.org/bin/bitcoin-core-27.0/bitcoin-27.0-x86_64-linux-gnu.tar.gz
+    - name: Unpack Bitcoin
+      run: tar -xzvf bitcoin-27.0-x86_64-linux-gnu.tar.gz
+
+    - name: Start Bitcoind
+      run: bitcoin-27.0/bin/bitcoind -regtest -rpcuser=admin -rpcpassword=admin -rpcport=18443 -fallbackfee=0.00001 -wallet=admin -txindex=1 &
+    - name: Create a wallet in Bitcoin regtest
+      run: bitcoin-27.0/bin/bitcoin-cli -regtest -rpcuser=admin -rpcpassword=admin -rpcport=18443 createwallet "admin"
+    - name: Create funds in Bitcoin regtest
+      run: bitcoin-27.0/bin/bitcoin-cli -regtest -rpcuser=admin -rpcpassword=admin -rpcport=18443 generatetoaddress 101 $(bitcoin-27.0/bin/bitcoin-cli -regtest -rpcuser=admin -rpcpassword=admin -rpcport=18443 getnewaddress)
+
+    - name: Build
+      run: RISC0_DEV_MODE=1 cargo build --verbose
+
+    - name: Create config overwrite file
+      run: |
```

Comment:
you can update this step like this for better readability

  run: |
    cat << EOF > /home/runner/overwrite.toml
    tracing_debug = "debug,bitcoincore_rpc=info,hyper=error"
    host = "127.0.0.1"
    port = 3000
    secret_key = "5555555555555555555555555555555555555555555555555555555555555555"
    verifiers_public_keys = [
      "4f355bdcb7cc0af728ef3cceb9615d90684bb5b2ca5f859ab0f0b704075871aa",
      "466d7fcae563e5cb09a0d1870bb580344804617879a14949cf22285f1bae3f27",
      "3c72addb4fdf09af94f0c94d7fe92a386a7e70cf8a1d85916386bb2535c7b1b1",
      "2c0b7cf95324a07d05398b240174dc0c2be444d96b159aa6c7f7b1e668680991",
      "9ac20335eb38768d2052be1dbbc3c8f6178407458e51e6b4ad22f1d91758895b"
    ]
     # ... rest of the configuration ...
     
    EOF